mage_ai/api/resources/PipelineResource.py

```diff
@@ -403,11 +403,19 @@ async def _add_markdown_block(block_doc: str, block_uuid: str, priority: int):
             if pipeline_doc:
                 await _add_markdown_block(pipeline_doc, self.model.uuid, 0)
 
+        pipeline_type = self.model.type
         await self.model.update(
             ignore_keys(payload, ['add_upstream_for_block_uuid']),
             update_content=update_content,
         )
-        switch_active_kernel(PIPELINE_TO_KERNEL_NAME[self.model.type])
+        try:
+            switch_active_kernel(PIPELINE_TO_KERNEL_NAME[self.model.type])
+        except Exception as e:
+            pipeline_type_updated = payload.get('type')
+            if pipeline_type_updated is not None:
```

Comment:
also check `pipeline_type_updated != orig_pipeline_type`

Reply:
@wangxiaoyou1993 I was gonna add that check but saw that we have one [here](https://github.com/mage-ai/mage-ai/blob/master/mage_ai/data_preparation/models/pipeline.py#L831-L836) already. Should we still check if the pipeline type is different from the original type here?

Reply:
that check includes many fields. if any of the field is different, set `should_save` to true.

the check here is to decide whether another DB update is needed.

Reply:
ah, got it. thx